std/algorithm/iteration.d

```diff
@@ -476,7 +483,7 @@ if (fun.length >= 1)
                 "Mapping function(s) must not return void: " ~ _funs.stringof);
         }
 
-        return MapResult!(_fun, Range)(r);
+        return MapResult!(_fun, Range)(move(r));
```

Comment:
If the point of moving here unconditionally is to have better performance, then a call to `move` has the exact opposite result because the template creates 2 temporaries, it calls 2 memcopies, and it calls a destructor (on an object with T.init state).